File: src/main/java/de/fluxparticle/animation/output/Log.java
package de.fluxparticle.animation.output;

public class Log {

	public static Log getLog(int level) {
		return new Log(level);
	}

	private final String indent;

	private Log(int level) {
		StringBuilder sb = new StringBuilder();
		for (int i = 0; i < level; i++) {
			sb.append(' ');
			sb.append(' ');
		}
		indent = sb.toString();
	}

	public void info(String line) {
//		System.out.println(indent + line);
	}

}

File: common/src/main/kotlin/de/fluxparticle/animation/output/Log.kt
package de.fluxparticle.animation.output

class Log private constructor(level: Int) {

    private val indent: String

    init {
        val sb = StringBuilder()
        for (i in 0 until level) {
            sb.append(' ')
            sb.append(' ')
        }
        indent = sb.toString()
    }

    fun info(line: String) {
        println(indent + line);
    }

    companion object {

        fun getLog(level: Int): Log {
            return Log(level)
        }
    }

}
